internal/common/objects/Audi.go

```diff
@@ -0,0 +1,67 @@
+package objects
+
+import (
+	phy "SOMAS2023/internal/common/physics"
+	"SOMAS2023/internal/common/utils"
+	"math"
+)
+
+type IAudi interface {
+	IPhysicsObject
+	UpdateGameState(state IGameState)
+}
+
+type Audi struct {
+	*PhysicsObject
+	target *MegaBike
+}
+
+// GetAudi is a constructor for Audi that initializes it with a new UUID and default position.
+func GetAudi() *Audi {
+	return &Audi{
+		PhysicsObject: GetPhysicsObject(utils.MassAudi),
+	}
+}
+
+func GetIAudi() IAudi {
+	return &Audi{
+		PhysicsObject: GetPhysicsObject(utils.MassAudi),
+	}
+}
+
+// Move as MegaBike, called by server
+func (audi *Audi) Move() {
+	if audi.target == nil { // no target, audi will stop
+		audi.velocity = 0.0
+	} else {
+		audi.velocity = 1.0 / audi.mass
+		audi.orientation = computeOrientation(audi.coordinates, audi.target.GetPosition())
+	}
+	audi.coordinates = phy.GetNewPosition(audi.coordinates, audi.velocity, audi.orientation)
+}
+
+func (audi *Audi) UpdateGameState(state IGameState) {
```

Comment:
I think that this is being confused with the internal state. The idea was that the server can pass a GameState object to each 'thing' every turn, in order to 'tell' it what the current game state is.

While this isn't wrong, I would suspect that the pattern is for this function to simply read as `audi.gameState = state`, and the computation can be done separately, i.e., `audi.ComputeTarget()`, say, using `audi.gameState`

Reply:
Added ComputeTarget Method